### ISSUE TITLE
arguments questions

### ISSUE THREAD

**Issue by city19992** [cmusatyalab/mega-nerf#1]
I read the paper about mega-nerf, which is amazing. Could I ask some questions like the arguments of the code? For example, what is the mean of '--chunk_paths $SCRATCH_PATH '? I do not see the notes of this parameter.

**Comment by city19992** [comment]
In addition, can I ask the author to add a small demo to show us how this program should be used?

**Comment by city19992** [comment]
In addition, may I ask the author to add a small demo to show us how this program should be used?

**Comment by hturki** [comment]
Hi - the chunks_path parameter is used as a list of scratch locations to write shuffled training data to disk. We should make this more clear in the README, but directly loading the full set of full-scale training images (around 2000 HD images per scene) into memory would be prohibitive, and so we write shuffled training batches into disk as a training step.

As mentioned in the README, training with against full scale data will write hundreds of GBs / several TBs of shuffled data to disk, but you can alternatively downsample the training data using train_scale_factor option. You could also use the train_every parameter to only use a subset of the training images. We'll provide better documentation of all the training parameters in the weeks to come.

Re: a small demo, are you asking for a sequence of example commands to train/render a scene? I tried putting some example usage in each section in the README (eg: ``python scripts/create_cluster_masks.py --config configs/mega-nerf/${DATASET_NAME}.yml --dataset_path $DATASET_PATH --output $MASK_PATH --grid_dim $GRID_X $GRID_Y`` in the training section). But perhaps this could be made more clear?

**Comment by city19992** [comment]
Thank you. I understand what the chunks_path parameter is used for, but how should I set the parameter? we notice that the program needs a document called 'metadata.pt'. However, we cannot find 'metadata.pt' in the project mega-nerf. Should I create a metadata.pt? if I need to create it, what should in this document?
We try to run mege-nerf on dataset Quad 6k. After generating the training partitions, I do not know how to set chunk_path.

**Comment by yuzewang1998** [comment]
Hi, thanks for your awesome work in NeRF fields.May I ask the question about the '--chunk_paths' and  '--exp_name' paramters? I dont know how to set those parameters too.
Thank you, looking forward to your reply.

**Comment by hturki** [comment]
@city19992 - the required metadata files should be included in the dataset links we provide. To recap the instructions in the README (https://github.com/cmusatyalab/mega-nerf/blob/main/README.md#quad-6k-dataset):

1) First download the raw Quad 6k dataset images. The folder should look something like this:
<img width="1706" alt="image" src="https://user-images.githubusercontent.com/868618/158466356-15ddcb85-4035-4c7d-8bf4-ff9eb8dbfd33.png">

2) Then download the dataset poses in the format that our repo expects from https://storage.cmusatyalab.org/mega-nerf-data/quad-pixsfm.tgz. The layout should look something like this (ie: everything is there except the raw images):

<img width="866" alt="image" src="https://user-images.githubusercontent.com/868618/158466894-40f3a6c6-0e99-4499-bac5-35159e0d72c7.png">

(For the Quad 6k dataset only we provide additional segmentation masks to avoid training pixels with transient objects during training)

3) Call ```python scripts/copy_images.py --image_path $RAW_PHOTO_PATH --dataset_path $CAMERA_POSE_PATH``` to copy the images into the directory layout that mega-nerf expects, where ```$RAW_PHOTO_PATH``` is the images directory in step 1 and ```$CAMERA_POSE_PATH``` is the directory in step 2

Note that we're still playing around with things a bit and you actually will want to redownload the directory in step 2 with the link that I just updated in the README (which has better camera poses and the segmentation masks that weren't present in the first upload). I'd recommend testing things out on either the Mill 19 or UrbanScene3D datasets vs Quad 6k at the moment if possible.

@novasky0709  - as previously mentioned:

> Hi - the chunks_path parameter is used as a list of scratch locations to write shuffled training data to disk. We should make this more clear in the README, but directly loading the full set of full-scale training images (around 2000 HD images per scene) into memory would be prohibitive, and so we write shuffled training batches into disk as a training step.

So the idea is that you set the --chunks_path parameter to a directory (or multiple directories) with adequate scratch space. Alternatively, you could use --dataset_type memory to keep everything in memory, but you might then need to set --train_scale_factor or --train_every to downsample or subsample the training set.
The --exp_name parameter is just the directory where we write the model checkpoints and tensorboard logs to. 

 Let me know if anything is still unclear!

**Comment by yuzewang1998** [comment]
@hturki Hi,thank you for your detailed reply. 
I think maybe @city19992 do the same wrong settings with me. I created an empty directory first and set it to the '--chunk_paths'.Therefore, I encountered the same error with @city19992 :
`[Errno 2] No such file or directory:  #'/home/slam/devdata/dataset/chunk/metadata.pt'`
The right thing to do is setting the --chunks_path parameter to a directory ( this directory doesn't exist) with adequate scratch space .Is it right? The training script seems to be working properly now.

Thank you again for your awesome work and detailed reply!

**Comment by hturki** [comment]
Hi @novasky0709 - thanks for clarifying. You're totally right - the directory specified by chunks_path should not initally exist (it's fine to reuse that directory after writing chunks to disk the first time). I had added some guards so that one would not read from a partially written chunks directory (we write metadata.pt at the very end) but I can see why this behavior is not intuitive. I'll leave this issue open until I push a commit that gives a more informative error message.

**Comment by Wanter-Z** [comment]
> @hturki Hi,thank you for your detailed reply. I think maybe @city19992 do the same wrong settings with me. I created an empty directory first and set it to the '--chunk_paths'.Therefore, I encountered the same error with @city19992 : `[Errno 2] No such file or directory: #'/home/slam/devdata/dataset/chunk/metadata.pt'` The right thing to do is setting the --chunks_path parameter to a directory ( this directory doesn't exist) with adequate scratch space .Is it right? The training script seems to be working properly now.
> 
> Thank you again for your awesome work and detailed reply!

Thank you for the guidance. It's the right way to debug the error mentioned above.

**Comment by hturki** [comment]
I added a hopefully more informative error message and some descriptions to all of the configuration options.
